pkg/sloop/webserver/debug.go

```diff
@@ -162,6 +164,107 @@ func listKeysHandler(tables typed.Tables) http.HandlerFunc {
 	}
 }
 
+type SloopKeyInfo struct {
+	TotalKeys     int64
+	EstimatedSize int64
+}
+
+type SloopKey struct {
+	TableName   string
+	PartitionID string
+}
+
+type histogram struct {
+	HistogramMap map[SloopKey]*SloopKeyInfo
+	TotalKeys    int
+	DeletedKeys  int
+}
+
+// returns IsDeletedExpired, TableName, PartitionId, EstimatedSize, error.
+func ParseSloopKey(item badgerwrap.Item) (bool, string, string, int64, error) {
+	key := item.Key()
+	parts := strings.Split(string(key), "/")
+	if len(parts) != 7 {
+		return false, "", "", 0, fmt.Errorf("key should have 6 parts: %x", key)
+	}
+
+	if parts[0] != "" {
+		return false, "", "", 0, fmt.Errorf("key should start with /: %x", key)
+	}
+
+	var isDeletedExpired = item.IsDeletedOrExpired()
+	var tableName = parts[1]
+	var partitionId = parts[2]
+	return isDeletedExpired, tableName, partitionId, item.EstimatedSize(), nil
+}
+
+func histogramHandler(tables typed.Tables) http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+
+		prefix := request.URL.Query().Get("prefix")
+		if len(prefix) > 0 {
```

Comment:
does it make sense to allow no prefix as a selection? would the result data set be large or slow?

Reply:
yes to analyze the whole data. It would be slow but as this is for debugging purposes only, so in my opinion it would be ok.